src/charm.py

```diff
@@ -175,13 +241,22 @@ def _request_kubelet_and_proxy_credentials(self):
         node_user = f"system:node:{kubernetes_snaps.get_node_name()}"
         self.kube_control.set_auth_request(node_user)
 
+    def _request_monitoring_token(self, event):
+        status.add(MaintenanceStatus("Requesting COS token"))
+        if not self._check_tokens_integration(event):
+            return
+
+        cos_user = f"system:cos:{kubernetes_snaps.get_node_name()}"
```

Comment:
same comment regarding `get_node_name`

it kinda makes me think the charm object itself should have it's own `self.get_node_name` so we don't make that mistake over and over